expression/evaluator_test.go

```diff
@@ -489,38 +442,31 @@ func (s *testEvaluatorSuite) TestBinopNumeric(c *C) {
 		{types.NewDecFromInt(10), ast.Mod, 0},
 	}
 
-	oldInSelectStmt := s.ctx.GetSessionVars().StmtCtx.InSelectStmt
-	s.ctx.GetSessionVars().StmtCtx.InSelectStmt = false
-	oldSQLMode := s.ctx.GetSessionVars().SQLMode
-	s.ctx.GetSessionVars().SQLMode |= mysql.ModeErrorForDivisionByZero
-	oldInInsertStmt := s.ctx.GetSessionVars().StmtCtx.InInsertStmt
-	s.ctx.GetSessionVars().StmtCtx.InInsertStmt = true
-	for _, t := range testcases {
-		fc := funcs[t.op]
-		f, err := fc.getFunction(s.ctx, datumsToConstants(types.MakeDatums(t.lhs, t.rhs)))
-		c.Assert(err, IsNil)
+	ctx.GetSessionVars().StmtCtx.InSelectStmt = false
+	ctx.GetSessionVars().SQLMode |= mysql.ModeErrorForDivisionByZero
+	ctx.GetSessionVars().StmtCtx.InInsertStmt = true
+	for _, tt := range testcases {
+		fc := funcs[tt.op]
+		f, err := fc.getFunction(ctx, datumsToConstants(types.MakeDatums(tt.lhs, tt.rhs)))
+		require.NoError(t, err)
 		_, err = evalBuiltinFunc(f, chunk.Row{})
-		c.Assert(err, NotNil)
+		require.Error(t, err)
 	}
 
-	oldDividedByZeroAsWarning := s.ctx.GetSessionVars().StmtCtx.DividedByZeroAsWarning
-	s.ctx.GetSessionVars().StmtCtx.DividedByZeroAsWarning = true
-	for _, t := range testcases {
-		fc := funcs[t.op]
-		f, err := fc.getFunction(s.ctx, datumsToConstants(types.MakeDatums(t.lhs, t.rhs)))
-		c.Assert(err, IsNil)
+	ctx.GetSessionVars().StmtCtx.DividedByZeroAsWarning = true
+	for _, tt := range testcases {
+		fc := funcs[tt.op]
+		f, err := fc.getFunction(ctx, datumsToConstants(types.MakeDatums(tt.lhs, tt.rhs)))
+		require.NoError(t, err)
 		v, err := evalBuiltinFunc(f, chunk.Row{})
-		c.Assert(err, IsNil)
-		c.Assert(v.Kind(), Equals, types.KindNull)
+		require.NoError(t, err)
+		require.Equal(t, types.KindNull, v.Kind())
 	}
-
-	s.ctx.GetSessionVars().StmtCtx.InSelectStmt = oldInSelectStmt
```

Comment:
Why not reset it?

Reply:
it doesn't reuse this ctx so that we don't have to reset it. for the next test, you create a new ctx to use.